src/charm.py

```diff
@@ -54,14 +55,23 @@ def __init__(self, *args):
         super().__init__(*args)
         self._container = self.unit.get_container(self._name)
         self._stored.set_default(agent_config="")
+        self._metrics_rules_src_path = os.path.join(self.charm_dir, METRICS_RULES_SRC_PATH)
+        self._metrics_rules_path = os.path.join(self.charm_dir, METRICS_RULES_PATH)
         self.service_patch = KubernetesServicePatch(
             self,
             [
                 (f"{self.app.name}-http-listen-port", self._http_listen_port),
                 (f"{self.app.name}-grpc-listen-port", self._grpc_listen_port),
             ],
         )
-        self._remote_write = PrometheusRemoteWriteConsumer(self)
+
+        if not os.path.isdir(self._metrics_rules_path):
+            shutil.copytree(
+                self._metrics_rules_src_path, self._metrics_rules_path, dirs_exist_ok=True
+            )
+        self._remote_write = PrometheusRemoteWriteConsumer(
+            self, alert_rules_path=self._metrics_rules_path
+        )
```

Comment:
`add_path` can be called multiple times
https://github.com/canonical/grafana-agent-k8s-operator/blob/a8f38b10fbbe5893d1c954ae22964af8fab65edf/lib/charms/prometheus_k8s/v0/prometheus_remote_write.py#L503

so maybe the constructor should take a list of paths instead of copytree?

Reply:
The (potentially custom) relation name should be passed in as well?

Reply:
The constructor does not support a list so while it would be a feature we could consider, it would not work here.

Reply:
> The constructor does not support a list so while it would be a feature we could consider, it would not work here.

I would also argue that we in that case would add a new function called `add_paths`, which would call `add_path` once per entry.